### PR TITLE
Role based middleware and token verification

### PR DESCRIPTION
**Pull request by shnawaz844**

### CODE REVIEW

File: src/common/decorators/roles.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { SetMetadata } from '@nestjs/common';
+import { Role } from '../enum/role.enum';
```
Comment (codevamp-technologies) [inline src/common/decorators/roles.decorator.ts:3]: there must be a empty line between nest imports and local imports

File: src/common/utility/responseFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export class ResponseFormatter {
```
Comment (codevamp-technologies) [inline src/common/utility/responseFormatter.ts:4]: add proper documentation of the class, what does the class do?

File: src/ticket_activities/ticket_activities.controller.ts [hunk 3]
```diff
@@ -5,35 +5,97 @@ import {
   Get,
   Param,
   Patch,
-  Delete,
+  UseGuards,
 } from '@nestjs/common';
 import { ActivitiesService } from './ticket_activities.service';
```
Comment (codevamp-technologies) [inline src/ticket_activities/ticket_activities.controller.ts:10]: empty line between local and global imports

File: src/users/users.controller.ts [hunk 4]
```diff
@@ -1,34 +1,108 @@
-import { Controller, Get, Post, Body, Patch, Param } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  UseGuards,
+} from '@nestjs/common';
```
Comment (codevamp-technologies) [inline src/users/users.controller.ts:9]: empty line between imports

File: src/users/users.controller.ts [hunk 5]
```diff
@@ -1,34 +1,108 @@
...
+import { RolesGuard } from 'src/common/guards/roles.guard';
+import { Roles } from 'src/common/decorators/roles.decorator';
+import { Role } from 'src/common/enum/role.enum';
 
 @Controller('api/user')
+@UseGuards(RolesGuard)
 export class UsersController {
   constructor(private readonly usersService: UsersService) {}
 
   @Post('signup')
```
Comment (codevamp-technologies) [inline src/users/users.controller.ts]: rename this to create

**Review by codevamp-technologies — CHANGES_REQUESTED** [review]
left comments

File: src/ticket_activities/ticket_activities.controller.ts (hunk 3 above)
Comment (shnawaz844) [reply src/ticket_activities/ticket_activities.controller.ts:10]: done

File: src/common/decorators/roles.decorator.ts (hunk 1 above)
Comment (shnawaz844) [reply src/common/decorators/roles.decorator.ts:3]: done

File: src/common/utility/responseFormatter.ts (hunk 2 above)
Comment (shnawaz844) [reply src/common/utility/responseFormatter.ts:4]: done

File: src/users/users.controller.ts (hunk 4 above)
Comment (shnawaz844) [reply src/users/users.controller.ts:9]: done

File: src/users/users.controller.ts (hunk 5 above)
Comment (shnawaz844) [reply src/users/users.controller.ts]: done

**Review by codevamp-technologies — APPROVED** [review]
looks good

**Pull request merged**